src/main/java/mate/academy/spring/controller/AuthenticationController.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.impl.response.UserResponseMapper;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.request.UserRequestDto;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.security.AuthenticationService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    private final UserResponseMapper userResponseMapper;
+
+    public AuthenticationController(AuthenticationService authenticationService,
+                                    UserResponseMapper userResponseMapper) {
+        this.authenticationService = authenticationService;
+        this.userResponseMapper = userResponseMapper;
+    }
```

Comment:
```suggestion
    private final AuthenticationService authenticationService;
    private final DtoResponseMapper<UserResponseDto, User> userResponseMapper;

    public AuthenticationController(AuthenticationService authenticationService,
                                    DtoResponseMapper<UserResponseDto, User> userResponseMapper) {
        this.authenticationService = authenticationService;
        this.userResponseMapper = userResponseMapper;
    }
```
we need to operate with abstractions